pkg/watcher/internal/metricsprovider/signalfx.go

```diff
@@ -206,3 +281,144 @@ func decodeMetricsPayload(payload interface{}) (float64, error) {
+    ....
+  },
+  "errors": []
+}
+
+https://dev.splunk.com/observability/reference/api/metrics_metadata/latest#endpoint-retrieve-metric-timeseries-metadata
+Sample metaData payload:
+{
+    "count": 5,
+    "partialCount": false,
+    "results": [
+        {
+            "active": true,
+            "created": 1614534848000,
+            "creator": null,
+            "dimensions": {
+                "host": "test.dev.com",
+                "sf_metric": null
+            },
+        "id": "EvVH6P7BgAA",
+		"lastUpdated": 0,
+		"lastUpdatedBy": null,
+		"metric": "cpu.utilization"
+       },
+       ....
+       ....
+	]
+}
+*/
+func getMetricsFromPayloads(metricData interface{}, metadata interface{}) (map[string]watcher.Metric, error) {
+	keyHostMap := make(map[string]string)
+	hostMetricMap := make(map[string]watcher.Metric)
+	if _, ok := metadata.(map[string]interface{}); !ok {
+		return hostMetricMap, fmt.Errorf("type conversion failed, found %T", metadata)
+	}
+	results := metadata.(map[string]interface{})["results"]
+	if results == nil {
+		return hostMetricMap, errors.New("unexpected payload: missing results field")
+	}
+
+	for _, v := range results.([]interface{}) {
+		_, ok := v.(map[string]interface{})
+		if !ok {
+			log.Errorf("type conversion failed, found %T", v)
+			continue
+		}
+		id := v.(map[string]interface{})["id"]
+		if id == nil {
+			log.Errorf("id not found in %v", v)
+			continue
+		}
+		_, ok = id.(string)
+		if !ok {
+			log.Errorf("id not expected type string, found %T", id)
+			continue
+		}
+		dimensions := v.(map[string]interface{})["dimensions"]
+		if dimensions == nil {
+			log.Errorf("no dimensions found in %v", v)
+			continue
+		}
+		_, ok = dimensions.(map[string]interface{})
+		if !ok {
+			log.Errorf("type conversion failed, found %T", dimensions)
+			continue
+		}
+		host := dimensions.(map[string]interface{})["host"]
+		if host == nil {
+			log.Errorf("no host found in %v", dimensions)
+			continue
+		}
+		if _, ok := host.(string); !ok {
+			log.Errorf("host not expected type string, found %T", host)
+		}
+		keyHostMap[id.(string)] = host.(string)
+	}
+
+	var data interface{}
+	data = metricData.(map[string]interface{})["data"]
+	if data == nil {
+		return hostMetricMap, errors.New("unexpected payload: missing data field")
+	}
+	keyMetricMap, ok := data.(map[string]interface{})
+	if !ok {
+		return hostMetricMap, errors.New("unable to deserialise data field")
+	}
+	for key, metric := range keyMetricMap {
+		if _, ok := keyHostMap[key]; !ok {
+			log.Errorf("no metadata found for key %v", key)
+			continue
+		}
+		values, ok := metric.([]interface{})
+		if !ok {
+			log.Errorf("unable to deserialise values for key %v", key)
+			continue
+		}
+		if len(values) == 0 {
+			log.Errorf("no metric value array could be decoded for key %v", key)
+			continue
+		}
+		// Find the average across returned values per 1 minute resolution
+		var sum float64
+		var count float64
+		for _, value := range values {
```

Comment:
For testing, we may want to break this arithmetic mean calculation into its own function but not a blocker for now.

Reply:
Sure will do it in future PR when I add tests https://github.com/paypal/load-watcher/issues/26